pkg/distributor/distributor.go

```diff
@@ -450,6 +452,28 @@ func (d *Distributor) Push(ctx context.Context, req *client.WriteRequest) (*clie
 	return &client.WriteResponse{}, lastPartialErr
 }
 
+func sortLabelsIfNeeded(labels []client.LabelAdapter) {
+	// no need to run sort.Slice, if labels are already sorted, which is most of the time.
+	// we can avoid extra memory allocations (mostly interface-related) this way.
+	sorted := true
+	last := ""
+	for _, l := range labels {
+		if strings.Compare(last, l.Name) > 0 {
+			sorted = false
```

Comment:
I think could be a bit simplified directly `return` from here. It will remove the `sorted` variable and the `if sorted` below.

Reply:
The logic is opposite though. If not sorted, then we don't return but sort.

Reply:
Damn, sure, you're right. Forgive my comment.